ethereum/core/src/main/java/org/hyperledger/besu/ethereum/forkid/ForkIdManager.java

```diff
@@ -135,16 +172,20 @@ public boolean peerCheck(final Bytes32 peerGenesisHash) {
   }
 
   private boolean isHashKnown(final Bytes forkHash) {
-    return forkIds.stream().map(ForkId::getHash).anyMatch(hash -> hash.equals(forkHash));
+    return Stream.concat(blockNumbersForkIds.stream(), timestampsForkIds.stream())
```

Comment:
seems we are doing a lot of concatenating.  we should just maintain a composed list of forks by number and timestamp